tensorflow_serving/batching/batching_util_test.cc

```diff
@@ -0,0 +1,93 @@
+/* Copyright 2017 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow_serving/batching/batching_util.h"
+
+
+
+#include <gtest/gtest.h>
+#include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/lib/core/status.h"
+#include "tensorflow/core/platform/types.h"
+
+namespace tensorflow {
+namespace serving {
+namespace {
+std::vector<std::pair<string, Tensor>> CreateInputsWithTensorShapes(
+        const std::vector<TensorShape>& shapes) {
+  std::vector<std::pair<string, Tensor>> inputs;
+  for (int i = 0; i < shapes.size(); ++i) {
+    inputs.push_back({"x" + std::to_string(i), Tensor(DT_FLOAT, shapes[i])});
+  }
+  return inputs;
+}
+
+
+TEST(BatchingUtilTest, CalculateMaxDimSizes) {
+  std::vector<TensorShape> shapes1;
+  shapes1.push_back({10, 20, 30});  // x0
+  shapes1.push_back({10, 100});  // x1
+  std::vector<std::pair<string, Tensor>> inputs1 =
+    CreateInputsWithTensorShapes(shapes1);
+  std::vector<TensorShape> shapes2;
+  shapes2.push_back({20, 50, 15});  // x0
+  shapes2.push_back({20, 101});  // x1
+  std::vector<std::pair<string, Tensor>> inputs2 =
+    CreateInputsWithTensorShapes(shapes2);
+  std::vector<std::vector<std::pair<string, Tensor>>> batch {inputs1, inputs2};
+  std::map<string, std::vector<int>> max_dim_sizes =
+    CalculateMaxDimSizes(batch);
+  std::map<string, std::vector<int>> true_max_dim_sizes;
+  true_max_dim_sizes["x0"] = {20, 50, 30};
+  true_max_dim_sizes["x1"] = {20, 101};
+  EXPECT_EQ(max_dim_sizes, true_max_dim_sizes);
```

Comment:
Please use the gmock matchers instead (they give more readable error messages if the test fails):
EXPECT_THAT(max_dim_sizes, UnorderedElementsAre(Pair("x0", ElementsAre(20,50,30), ...

Same elsewhere in your test code please.